spec/models/authenticator/httpd_spec.rb

```diff
@@ -190,6 +191,78 @@ def authenticate
       end
     end
 
+    context "with potential for multiple user records" do
+      let(:dn) { 'cn=sally,ou=people,ou=prod,dc=example,dc=com' }
+      let(:config) { {:httpd_role => true} }
+
+      let(:username) { 'saLLy' }
+      let(:user_groups) { 'wibble@fqdn:bubble@fqdn' }
+
+      let(:headers) do
+        super().merge('X-Remote-User-FullName'  => 'Sally Porsche',
+                      'X-Remote-User-FirstName' => 'Sally',
+                      'X-Remote-User-LastName'  => 'Porsche',
+                      'X-Remote-User-Email'     => 'Sally@example.com')
+      end
+
+      context "when user record with userid in upn format already exists" do
+        let!(:sally_username) { FactoryGirl.create(:user, :userid => 'sAlly') }
+        let!(:sally_dn) { FactoryGirl.create(:user, :userid => dn) }
+        let!(:sally_upn) { FactoryGirl.create(:user, :userid => 'sAlly@example.com') }
+
+        it "leaves user record with userid in username format unchanged" do
+          expect(-> { authenticate }).to_not change { sally_username.reload.userid }
+        end
+
+        it "leaves user record with userid in distinguished name format unchanged" do
+          expect(-> { authenticate }).to_not change { sally_dn.reload.userid }
+        end
+
+        it "downcases user record with userid in upn format" do
+          expect(-> { authenticate })
+            .to change { sally_upn.reload.userid }.from("sAlly@example.com").to("sally@example.com")
+        end
+      end
+
+      context "when user record with userid in upn format does not already exists" do
+        it "updates userid from username format to upn format" do
+          sally_username = FactoryGirl.create(:user, :userid => 'sally')
+          expect(-> { authenticate }).to change { sally_username.reload.userid }.from("sally").to("sally@example.com")
+        end
+
+        it "updates userid from distinguished name format to upn format" do
+          sally_dn = FactoryGirl.create(:user, :userid => dn)
+          expect(-> { authenticate }).to change { sally_dn.reload.userid }.from(dn).to("sally@example.com")
+        end
+
+        it "does not modify userid if already in upn format" do
+          sally_upn = FactoryGirl.create(:user, :userid => 'sally@example.com')
+          expect(-> { authenticate }).to_not change { sally_upn.reload.userid }
+        end
+      end
+
+      context "when user record is for a different region" do
+        let(:my_region_number) { Classification.my_region_number }
```

Comment:
Just curious, why get the current region from `Classification` and not the one in `ApplicationRecord` since `Classification` is not related to any of this?

Same for the next 2 lines